code/game/machinery/fax_machine.dm

```diff
@@ -164,73 +175,59 @@ var/list/alldepartments = list()
 
 	switch(action)
 		if("send")
-			if(tofax)
-				if(target_department == DEPARTMENT_HC)
-					highcom_fax(src, tofax.info, tofax.name, usr)
-					fax_cooldown = 600
-
-				else if(target_department == DEPARTMENT_PROVOST)
-					provost_fax(src, tofax.info, tofax.name, usr)
-					fax_cooldown = 600
-
-				else if(target_department == DEPARTMENT_CMB)
-					cmb_fax(src, tofax.info, tofax.name, usr)
-					fax_cooldown = 600
-
-				else if(target_department == DEPARTMENT_WY)
-					company_fax(src, tofax.info, tofax.name, usr)
-					fax_cooldown = 600
-
-				else if(target_department == DEPARTMENT_PRESS)
-					press_fax(src, tofax.info, tofax.name, usr)
-					fax_cooldown = 600
-
-				else
-					general_fax(src, tofax.info, tofax.name, usr)
-					fax_cooldown = 600
-
-				COOLDOWN_START(src, send_cooldown, fax_cooldown)
-				SendFax(tofax.info, tofax.name, usr, target_department, network, src)
-				to_chat(usr, "Message transmitted successfully.")
-				. = TRUE
+			if(!original_fax)
+				to_chat(ui.user, SPAN_NOTICE("No paper loaded"))
+				return
+
+			copy_fax_paper()
+
+			outgoing_fax_message(ui.user)
+
+			COOLDOWN_START(src, send_cooldown, fax_cooldown)
+			to_chat(ui.user, "Message transmitted successfully.")
+			. = TRUE
 
 		if("ejectpaper")
-			if(tofax)
-				if(!ishuman(usr))
-					to_chat(usr, SPAN_WARNING("You can't do it."))
-				else
-					tofax.forceMove(usr.loc)
-					usr.put_in_hands(tofax)
-					to_chat(usr, SPAN_NOTICE("You take \the [tofax] out of \the [src]."))
-					tofax = null
-				. = TRUE
+			if(!original_fax)
+				to_chat(ui.user, SPAN_NOTICE("No paper loaded"))
+			if(!ishuman(ui.user))
+				to_chat(ui.user, SPAN_NOTICE("You can't do that."))
+				return
+
+			original_fax.forceMove(ui.user.loc)
+			ui.user.put_in_hands(original_fax)
+			to_chat(ui.user, SPAN_NOTICE("You take \the [original_fax.name] out of \the [src]."))
+			original_fax = null
+			. = TRUE
 
 		if("insertpaper")
-			var/obj/item/I = usr.get_active_hand()
-			if(istype(I, /obj/item/paper))
-				usr.drop_inv_item_to_loc(I, src)
-				tofax = I
-				to_chat(usr, SPAN_NOTICE("You put \the [tofax] into \the [src]."))
+			// Repeating code? This is not ideal. Why not put this functionality inside of a proc?
+			var/obj/item/I = ui.user.get_active_hand()
+			if(istype(I, /obj/item/paper) || istype(I, /obj/item/paper_bundle || istype(I, /obj/item/photo)))
+				ui.user.drop_inv_item_to_loc(I, src)
+				original_fax = I
+				to_chat(ui.user, SPAN_NOTICE("You put \the [original_fax.name] into \the [src]."))
```

Comment:
Make the fax machine jam if more than 5 items are inserted in a bundle (different to_chat, maybe a sfx, disable send_fax (just need to check the length of the bundle if its a bundle).